section-07-ci-and-publishing/model-package/tox.ini

```diff
@@ -0,0 +1,107 @@
+# Tox is a generic virtualenv management and test command line tool. Its goal is to
+# standardize testing in Python. We will be using it extensively in this course.
+
+# Using Tox we can (on multiple operating systems):
+# + Eliminate PYTHONPATH challenges when running scripts/tests
+# + Eliminate virtualenv setup confusion
+# + Streamline steps such as model training, model publishing
+
+
+[tox]
+envlist = test_package, typechecks, stylechecks, lint
+skipsdist = True
+
+[testenv]
+install_command = pip install {opts} {packages}
+
+passenv =
+	KAGGLE_USERNAME
+	KAGGLE_KEY
+	GEMFURY_PUSH_URL
+
+[testenv:test_package]
+deps =
+	-rrequirements/test_requirements.txt
+
+setenv =
+	PYTHONPATH=.
+	PYTHONHASHSEED=0
+
+commands=
+	python regression_model/train_pipeline.py
+	pytest \
+	-s \
+	-vv \
+	{posargs:tests/}
+
+[testenv:train]
+envdir = {toxworkdir}/test_package
+deps =
+	{[testenv:test_package]deps}
+
+setenv =
+	{[testenv:test_package]setenv}
+
+commands=
+	python regression_model/train_pipeline.py
+
+[testenv:fetch_data]
+envdir = {toxworkdir}/test_package
+deps =
+	{[testenv:test_package]deps}
+
+setenv =
+	{[testenv:test_package]setenv}
+
+commands=
+	# fetch
+	kaggle competitions download -c house-prices-advanced-regression-techniques -p ./regression_model/datasets
+	# unzip
+	unzip ./regression_model/datasets/house-prices-advanced-regression-techniques.zip -d ./regression_model/datasets
+
+
+[testenv:publish_model]
```

Comment:
new command to publish the model in ci (but students can run locally if they want to test)